### ISSUE TITLE
add ReadOnlySpan<char> as string parameter

### ISSUE THREAD

**Issue by jpbruyere** [xin9le/FastEnum#9]
First, thank you for this remarkable utility. Do you plan to introduce 'ReadOnlySpan<char>' as string parameter?

**Comment by xin9le** [comment]
Thank you for your good suggestion! Just to confirm, do you mean you want some overload methods like following?

- `bool IsDefined<T>(ReadOnlySpan<char> value)`
- `T Parse<T>(ReadOnlySpan<char> value)`
- `bool TryParse<T>(ReadOnlySpan<char> value, out T result)`
- `bool TryParse<T>(ReadOnlySpan<char> value, bool ignoreCase, out T result)`

**Comment by jpbruyere** [comment]
Absolutely, I guess every living net library will get to this point soon.

**Comment by xin9le** [comment]
I think I would love to implement `ReadOnlySpan<char>` overloads. However `ReadOnlySpan<char>.GetHashCode()` throws `NotSupportedException`, so I have to workaround this problem. I'm not sure I can do it well, but I'll try it.

**Comment by jpbruyere** [comment]
Have you consider [this method](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.compareinfo.gethashcode?view=net-5.0#System_Globalization_CompareInfo_GetHashCode_System_ReadOnlySpan_System_Char__System_Globalization_CompareOptions_):

```csharp
public int GetHashCode (ReadOnlySpan<char> source, System.Globalization.CompareOptions options);
```

**Comment by lsoft** [comment]
@xin9le I would love this feature too! :)

**Comment by xin9le** [comment]
I am motivated to improve, but I have been working extremely hard for the past few years. I apologize for putting the improvement task on the back burner.

**Comment by xin9le** [comment]
Although it has taken a considerable amount of time, we have now addressed this request. We will be releasing the NuGet Package in the near future 🚀

**Comment by xin9le** [comment]
I've just released [v2.0.0-preview2](https://github.com/xin9le/FastEnum/releases/tag/v2.0.0-preview2). Please take advantage of it.
